Source/AssetRipper.Processing/AnimationClips/CustomCurveResolver.cs

```diff
@@ -20,505 +19,328 @@ public CustomCurveResolver(IAnimationClip clip)
-							case 4:
-								return "m_RotationOffset.x";
-							case 5:
-								return "m_RotationOffset.y";
-							case 6:
-								return "m_RotationOffset.z";
-							case 7:
-								return "m_AffectRotationX";
-							case 8:
-								return "m_AffectRotationY";
-							case 9:
-								return "m_AffectRotationZ";
-							case 10:
-								return "m_Active";
-							case 11:
-								return $"m_Sources.Array.data[{attribute >> 8}].sourceTransform";
-							case 12:
-								return $"m_Sources.Array.data[{attribute >> 8}].weight";
-						}
+						// Not Crc
+
+						uint property = attribute & 0xF; // 0...1111 (max 15)
+
+						return property switch
+						{
+							0u  => "m_RestRotation.x",
+							1u  => "m_RestRotation.y",
+							2u  => "m_RestRotation.z",
+							3u  => "m_Weight",
+							4u  => "m_RotationOffset.x",
+							5u  => "m_RotationOffset.y",
+							6u  => "m_RotationOffset.z",
+							7u  => "m_AffectRotationX",
+							8u  => "m_AffectRotationY",
+							9u  => "m_AffectRotationZ",
+							10u => "m_Active",
+							11u => $"m_Sources.Array.data[{attribute >> 8}].sourceTransform",
+							12u => $"m_Sources.Array.data[{attribute >> 8}].weight",
+
+							_ => throw new ArgumentException($"Unknown attribute {attribute} for {type}"),
+						};
 					}
-					throw new ArgumentException($"Unknown attribute {attribute} for {type}");
 
-#warning TODO:
 				case BindingCustomType.ScaleConstraint:
 					{
-						uint property = attribute & 0xF;
-						switch (property)
-						{
-							case 0:
-								return "m_ScaleAtRest.x";
-							case 1:
-								return "m_ScaleAtRest.y";
-							case 2:
-								return "m_ScaleAtRest.z";
-							case 3:
-								return "m_Weight";
-							case 4:
-								return "m_ScalingOffset.x";
-							case 5:
-								return "m_ScalingOffset.y";
-							case 6:
-								return "m_ScalingOffset.z";
-							case 7:
-								return "m_AffectScalingX";
-							case 8:
-								return "m_AffectScalingY";
-							case 9:
-								return "m_AffectScalingZ";
-							case 10:
-								return "m_Active";
-							case 11:
-								return $"m_Sources.Array.data[{attribute >> 8}].sourceTransform";
-							case 12:
-								return $"m_Sources.Array.data[{attribute >> 8}].weight";
-						}
+						// Not Crc
+
+						uint property = attribute & 0xF; // 0...1111 (max 15)
+
+						return property switch
+						{
+							0u  => "m_ScaleAtRest.x",
+							1u  => "m_ScaleAtRest.y",
+							2u  => "m_ScaleAtRest.z",
+							3u  => "m_Weight",
+							4u  => "m_ScalingOffset.x",
+							5u  => "m_ScalingOffset.y",
+							6u  => "m_ScalingOffset.z",
+							7u  => "m_AffectScalingX",
+							8u  => "m_AffectScalingY",
+							9u  => "m_AffectScalingZ",
+							10u => "m_Active",
+							11u => $"m_Sources.Array.data[{attribute >> 8}].sourceTransform",
+							12u => $"m_Sources.Array.data[{attribute >> 8}].weight",
+
+							_ => throw new ArgumentException($"Unknown attribute {attribute} for {type}"),
+						};
```

Comment:
This is an exception to my formatting comment above. I generally like looking at switch expressions significantly more than switch statements.